torch/distributions/exp_family.py

```diff
@@ -58,3 +58,14 @@ def entropy(self):
         for np, g in zip(nparams, gradients):
             result -= np * g
         return result
+
+    @staticmethod
+    def _from_natural_params(*params):
+        raise NotImplementedError
+
+    def product_along_axis(self, dim=-1, keepdim=True):
```

Comment:
Looks great! We just need a good docstring here. It's probably worth mentioning the product-of-experts model and the equation
```
For a batched distribution d and a single observation x::

    d.product_along_axis(d).log_prob(x) = d.log_prob(x).sum(d)
```

Reply:
I guess you could even add this as a unit test :smile:

Reply:
OK, will do. Do you think `prod` is a better name? e.g. consistent with torch.prod?

Reply:
I don't know the best name. We want to avoid confusion between operations on density (your `.prod()` here) and operations on samples of the distribution (e.g. the product of a Gamma r.v. and centered Normal r.v. is another Normal r.v.). Maybe
- `.prod()`?
- `.log_prob_sum()`?

Reply:
`d.product_along_axis().log_prob(x) = d.log_prob(x).sum()` doesn't hold since the LHS is a normalized product :/ This makes me think that it makes more sense to include "normalized" in the function name. So maybe: `normalized_prod`?

Reply:
Hmm, `.normalized_prod()` seems fine. Or maybe `.fuse()`, as this is classic Bayesian fusion?

Reply:
I haven't really heard of Bayesian fusion, is it used in the data fusion context? I think `normalized_product()` is concise.

Reply:
Yes, by "Bayesian fusion" I mean "the core Bayesian operation in sensor fusion" or "data fusion" or "information fusion". I.e. if each sensor is an expert then sensor fusion is product of experts.